### PR TITLE
boards: arm: nrf9161dk_nrf9161: Add ext flash support for 0.7.0 version

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add support for external flash on the 0.7.0 DK version.

### CODE REVIEW

File: boards/arm/nrf9161dk_nrf9161/revision.cmake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+board_check_revision(
+  FORMAT MAJOR.MINOR.PATCH
+  DEFAULT_REVISION 0.7.1
```
Comment (maxd-nordic) [inline boards/arm/nrf9161dk_nrf9161/revision.cmake:6]: This should be the version that is actually sent to customers. Not some old prototype version.

Comment (eivindj-nordic) [reply boards/arm/nrf9161dk_nrf9161/revision.cmake]: Thanks, updated to the latest 0.9.1

File: boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common_0_7_0.dtsi [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		size = <268435456>;
+		has-dpd;
+		t-enter-dpd = <3000>;
+		t-exit-dpd = <30000>;
+	};
+};
+
+/ {
+	 chosen {
+		 nordic,pm-ext-flash = &gd25lb256;
```
Comment (anangl) [inline boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common_0_7_0.dtsi:29]: Why is this set upstream?

Comment (eivindj-nordic) [reply boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common_0_7_0.dtsi]: Removed, thanks.

**Review by anangl — APPROVED** [review]

**Review by maxd-nordic — APPROVED** [review]

**Comment by eivindj-nordic** [comment]
This PR adds back support for the external flash on the nRF9161DK 0.7.0 board revision that was removed in #62681. This support is still needed as the nRF9161DK version 0.7.0 is still in use.

**Comment by anangl** [comment]
> This PR adds back support for the external flash on the nRF9161DK 0.7.0 board revision that was removed in #62681. This support is still needed as the nRF9161DK version 0.7.0 is still in use.

I think it would be good to correct this still in 3.5.0.

@carlescufi @jhedberg

**Review by carlescufi — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**